runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/state/GrpcStateService.java

```diff
@@ -31,10 +31,16 @@
 /** An implementation of the Beam Fn State service. */
 public class GrpcStateService extends BeamFnStateGrpc.BeamFnStateImplBase
     implements StateDelegator, FnService {
+  /**
+   * Create a new {@link GrpcStateService}.
+   */
+  public static GrpcStateService create() {
+    return new GrpcStateService();
+  }
+
   private final ConcurrentHashMap<String, StateRequestHandler> requestHandlers;
 
-  public GrpcStateService()
-      throws Exception {
+  public GrpcStateService() {
```

Comment:
Could you explain the motivation for this change? Usually this is done in order to make the regular constructor private. If we're leaving that exposed, I'm not sure what the static method adds.

Reply:
The static constructor is to make it consistent, stylistically, with the rest of the services.

I'm not removing the public constructor to avoid forcing all of the flink commit history to change as a result, but if it's NBD then I remove its access also.

Reply:
It's not currently being used by Flink in master. The service management bits are being reimplemented either way, so it's safe to go ahead and remove if that's the only reason you've kept this.